src/runtime/audience-action-flow.js

```diff
@@ -311,10 +311,31 @@ export class AudienceActionFlow {
       this.storage_.storeEvent(StorageKeys.SURVEY_DATA_TRANSFER_FAILED);
     }
     const surveyDataTransferResponse = new SurveyDataTransferResponse();
-    surveyDataTransferResponse.setSuccess(gaLoggingSuccess);
+    surveyDataTransferResponse.setSuccess(dataTransferSuccess);
     this.activityIframeView_.execute(surveyDataTransferResponse);
   }
 
+  /**
+   * Attempts to log survey data.
+   * @param {SurveyDataTransferRequest} request
+   * @return {boolean}
+   * @private
+   */
+  async attemptSurveyDataTransfer(request) {
+    // @TODO(justinchou): execute callback with setOnInterventionComplete
+    // then check for success
+    const {onResult} = this.params_;
+    if (onResult) {
+      try {
+        return onResult(request);
```

Comment:
I think this should `await` the onResult call, in case it returns a promise

Reply:
It could return a promise that eventually rejects

Reply:
Done.